habitat-lab/habitat/robots/robot_interface.py

```diff
@@ -38,3 +40,54 @@ def get_link_and_joint_names(self) -> str:
         for link_id in self.sim_obj.get_link_ids():
             link_joint_names += f"{link_id} = {self.sim_obj.get_link_name(link_id)} | {self.sim_obj.get_link_joint_name(link_id)} :: type = {self.sim_obj.get_link_joint_type(link_id)} \n"
         return link_joint_names
+
+    def _update_motor_settings_cache(self):
+        """Updates the JointMotorSettings cache for cheaper future updates"""
+        self.joint_motors = {}
+        for (
+            motor_id,
+            joint_id,
+        ) in self.sim_obj.existing_joint_motor_ids.items():
+            self.joint_motors[joint_id] = (
+                motor_id,
+                self.sim_obj.get_joint_motor_settings(motor_id),
+            )
+
+    def _validate_joint_idx(self, joint):
+        if joint not in self.joint_motors:
+            raise ValueError(
+                f"Requested joint {joint} not in joint motors with indices (keys {self.joint_motors.keys()}) and {self.joint_motors}"
+            )
+
+    def _get_motor_pos(self, joint):
+        self._validate_joint_idx(joint)
+        return self.joint_motors[joint][1].position_target
+
+    def _set_motor_pos(self, joint, ctrl):
+        self._validate_joint_idx(joint)
+        self.joint_motors[joint][1].position_target = ctrl
+        self.sim_obj.update_joint_motor(
+            self.joint_motors[joint][0], self.joint_motors[joint][1]
+        )
+
+    def _vector_to_plane_proj(self, u, n):
```

Comment:
These functions should probably be moved to a util module like[ geometry_utils.py](https://github.com/facebookresearch/habitat-lab/blob/main/habitat-lab/habitat/utils/geometry_utils.py)

Reply:
Remove the projection code. Thank you!